components/pd_client/src/util.rs

```diff
@@ -472,19 +472,26 @@ where
     }
 }
 
+pub fn call_option_inner(inner: &Inner) -> CallOption {
+    inner
+        .target_info()
+        .call_option()
+        .timeout(Duration::from_secs(REQUEST_TIMEOUT))
+}
+
 /// Do a request in synchronized fashion.
 pub fn sync_request<F, R>(client: &Client, mut retry: usize, func: F) -> Result<R>
 where
-    F: Fn(&PdClientStub) -> GrpcResult<R>,
+    F: Fn(&PdClientStub, CallOption) -> GrpcResult<R>,
 {
     loop {
         let ret = {
             // Drop the read lock immediately to prevent the deadlock between the caller
             // thread which may hold the read lock and wait for PD client thread
             // completing the request and the PD client thread which may block
             // on acquiring the write lock.
-            let client_stub = client.inner.rl().client_stub.clone();
```

Comment:
Did you mean the read lock here is stil held while getting `CallOption`?

Reply:
The read lock guard is dropped because of `clone`.

Reply:
yes, I suspect this from the stack trace, the back ground task has wait for the futex ?